lib/screens/place_detail/place_detail_screen.dart

```diff
@@ -1,15 +1,228 @@
-import 'package:cambodia_geography/widgets/cg_app_bar_title.dart';
+import 'package:cambodia_geography/cambodia_geography.dart';
+import 'package:cambodia_geography/constants/config_constant.dart';
+import 'package:cambodia_geography/exports/widgets_exports.dart';
+import 'package:cambodia_geography/helper/number_helper.dart';
+import 'package:cambodia_geography/mixins/cg_theme_mixin.dart';
+import 'package:cambodia_geography/models/places/place_model.dart';
+import 'package:cambodia_geography/screens/place_detail/local_widgets/place_title.dart';
+import 'package:cambodia_geography/widgets/cg_image_view.dart';
 import 'package:flutter/material.dart';
-import 'package:swipeable_page_route/swipeable_page_route.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+import 'package:page_indicator/page_indicator.dart';
 
-class PlaceDetailScreen extends StatelessWidget {
-  const PlaceDetailScreen({Key? key}) : super(key: key);
+class PlaceDetailScreen extends StatefulWidget {
+  const PlaceDetailScreen({
+    required this.place,
+    Key? key,
+  }) : super(key: key);
+
+  final PlaceModel place;
+
+  @override
+  _PlaceDetailScreenState createState() => _PlaceDetailScreenState();
+}
+
+class _PlaceDetailScreenState extends State<PlaceDetailScreen> with CgThemeMixin {
+  late ScrollController scrollController;
+  late PageController pageController;
+  late bool isShow;
+  late CambodiaGeography geo;
+  @override
+  void initState() {
+    scrollController = ScrollController();
+    pageController = PageController();
+    geo = CambodiaGeography.instance;
+    isShow = false;
+    super.initState();
+  }
 
   @override
   Widget build(BuildContext context) {
+    PlaceModel place = widget.place;
+    final double expandedHeight = MediaQuery.of(context).size.width / 2;
+    scrollController.addListener(() {
+      if (scrollController.offset >= expandedHeight)
```

Comment:
don't add a listener here, add a listener is mean that you are a listener to a list, when build It will continuously add to controller.

Add it only once on init state.